### PR TITLE
586 expandable card

### PR DESCRIPTION
**Pull request by 0xRowdy**
## GitHub Issue

[GitHub issue.](https://github.com/HausDAO/daohaus-monorepo/issues/586)

## Changes

Created a Card with a trigger that allows the card to display the hidden content.

image

## Packages Added

@radix-ui/react-collapsible

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: libs/ui/src/components/atoms/Tooltip/Tooltip.styles.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { border } from '../../../theme/global/border';
 export const TooltipProvider = TooltipPrimitive.Provider;
 export const TooltipRoot = TooltipPrimitive.Root;
 export const TooltipTrigger = styled(TooltipPrimitive.Trigger)`
-  background-color: ${({ theme }: { theme: Theme }) => theme.bgColor};
+  background-color: transparent;
```
Comment (brossetti1) [inline libs/ui/src/components/atoms/Tooltip/Tooltip.styles.ts:12]: just checking - was this suppose to be changed?

Comment (0xRowdy) [reply libs/ui/src/components/atoms/Tooltip/Tooltip.styles.ts:12]: Yeah, the background was colored with the primary theme. It made it so there was a small box that was a different color than the card. It makes more sense to have the background be the color of the parent.

File: libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+export const StyledCollapsibleContent = styled(Collapsible.Content)`
+  /* TODO Animations? */
+  /* &[data-state='open'] {
+    animation: ${open} 300ms ease-out;
+  }
+  &[data-state='closed'] {
+    animation: ${close} 300ms ease-out;
+  } */
+`;
```
Comment (brossetti1) [inline libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts:60]: seems like a good idea, the animation here is simple and goes along way to making interactions more pleasnt

File: libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  onChange,
+  width = 'fit-content',
+  children,
+  collapsibleContent,
+}: CardProps) => {
+  return (
+    <StyledCollapsibleCard open={open} onChange={onChange} width={width}>
+      <InnerCard width={'90%'}>
+        {children}
+        <StyledCollapsibleCardTrigger asChild>
```
Comment (brossetti1) [inline libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.tsx:31]: is `asChild` something in our library or is that coming from radix?

File: libs/ui/src/styled.d.ts [hunk 4]
```diff
@@ -68,6 +68,16 @@ declare module 'styled-components' {
...
+    collapsibleCard: {
+      bg: string;
+      bgInnerCard: string;
+      border: string;
+      borderInnerCard: string;
+      hoverBg: string;
+      hoverBorder: string;
+      focusBg: string;
+      focusBorder: string;
+    };
```
Comment (brossetti1) [inline libs/ui/src/styled.d.ts:80]: so not sure where this config comes from, only ones I am seeing being used are `borderInnerCard` and `bgInnerCard`

looking in the file i see some of these repeated so not sure what the purpose is - are the unused attributes for custom themeing for the daos in the ui?

File: libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+const open = keyframes({
+  from: { height: 0 },
+  to: { height: 100 },
+});
+
+const close = keyframes({
+  from: { height: 100 },
+  to: { height: 0 },
+});
```
Comment (brossetti1) [inline libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts:50]: should these go into https://github.com/HausDAO/daohaus-monorepo/blob/develop/libs/ui/src/animations/general.ts#L15?

**Review by brossetti1 — APPROVED** [review]
left a few questions, mainly getting used to the UI library now. Not the best at css so cant help out there too much -- looks good otherwise

File: libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts (hunk 2 above)
Comment (0xRowdy) [reply libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts:60]: Good call.

File: libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+          <StyledCollapsibleCardButton>
+            Expand Me
```
[the remark this answers is not in the document]
Comment (0xRowdy) [reply libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.tsx:33]: No this should be a prop. Ty

File: libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.tsx (hunk 3 above)
Comment (0xRowdy) [reply libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.tsx:31]: asChild is implemented by radix. It allows for us to override the element radix uses. In this example redux uses buttons for triggers. We could use a div or SVG or whatever. We use a button but I don't want a button inside a button or to restyle the redux trigger button.

File: libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts (hunk 2 above)
Comment (0xRowdy) [reply libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts:60]: Fixed

File: libs/ui/src/styled.d.ts (hunk 4 above)
Comment (0xRowdy) [reply libs/ui/src/styled.d.ts:80]: Cleaned up

File: libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts (hunk 5 above)
Comment (0xRowdy) [reply libs/ui/src/components/molecules/CollapsibleCard/CollapsibleCard.styles.ts:50]: Finished

**Pull request merged**